### PR TITLE
[Pfapack] Fix .so Contents

### PR DESCRIPTION
**Pull request by xrq-phys**
- `-force-load` only work for LLVM-based linkers. Fallback to `--whole-archive`
- Update upstream to use a CMakeLists.txt with all [sdcz] routines

### CODE REVIEW

File: P/Pfapack/build_tarballs.jl [hunk 1]
```diff
@@ -25,7 +26,7 @@ make install
 
 # Manual conversion from static to dynamic lib.
 # Avoid linking libgfortran (these .f files make no reference to Fortran libs.)
-cc -shared -Wl,-force_load,${prefix}/lib/libpfapack.a \
+cc -shared -Wl,--whole-archive ${prefix}/lib/libpfapack.a -Wl,--no-whole-archive \
```
Comment (giordano) [inline P/Pfapack/build_tarballs.jl:29]:
```suggestion
cc -shared $(flagon -Wl,--whole-archive) ${prefix}/lib/libpfapack.a $(flagon -Wl,--no-whole-archive) \
```

File: P/Pfapack/build_tarballs.jl [hunk 2]
```diff
@@ -17,15 +17,27 @@ cd $WORKSPACE/srcdir
...
+case ${target} in
+
+    *"apple"*)
+        # Apple implements a different linker option.
+        export A2SO="-Wl,-force_load,${prefix}/lib/libpfapack.a"
+        ;;
+    *)
+        export A2SO="-Wl,--whole-archive ${prefix}/lib/libpfapack.a -Wl,--no-whole-archive"
+        ;; 
+esac
```
Comment (giordano) [inline P/Pfapack/build_tarballs.jl:36]: `flagon` does this for you: https://github.com/JuliaPackaging/Yggdrasil/pull/4784#discussion_r853849184

Comment (xrq-phys) [reply P/Pfapack/build_tarballs.jl]: I see. Thanks!

Comment (giordano) [reply P/Pfapack/build_tarballs.jl]: If you're curious, this is the script: https://github.com/JuliaPackaging/Yggdrasil/blob/acb37f70728e86b5f747f752dbab71a1ce198fb3/0_RootFS/Rootfs/bundled/utils/flagon

Comment (xrq-phys) [reply P/Pfapack/build_tarballs.jl]: Impressive. Thanks.

**Pull request merged**